core/mock_test.go

```diff
@@ -59,13 +61,16 @@ type mockEC2 struct {
 	dltverr error
 }
 
-func (m mockEC2) DescribeSpotPriceHistory(in *ec2.DescribeSpotPriceHistoryInput) (*ec2.DescribeSpotPriceHistoryOutput, error) {
-	return m.dspho, m.dspherr
+func (m mockEC2) DescribeSpotPriceHistoryPages(in *ec2.DescribeSpotPriceHistoryInput, f func(*ec2.DescribeSpotPriceHistoryOutput, bool) bool) error {
+	for i, page := range m.dsphpo {
+		f(page, i == len(m.dsphpo)-1)
+	}
+	return m.dsphperr
 }
 
 func (m mockEC2) DescribeInstancesPages(in *ec2.DescribeInstancesInput, f func(*ec2.DescribeInstancesOutput, bool) bool) error {
 	f(m.dio, true)
```

Comment:
It looks like all existing `Pages` tests only test with a single page? We should think about changing that.